src/language/dynamics/transition/Transition.re

```diff
@@ -832,19 +832,31 @@ module Transition = (EV: EV_MODE) => {
       let. _ = otherwise(env, d);
       let.wrap_closure _ = env;
       Indet;
-    | Asc(d, t) =>
-      let. _ = otherwise(env, d => Asc(d, t) |> rewrap)
-      and. d' = req_final(req(state, env), d => Asc(d, t) |> wrap_ctx, d);
-      switch (Ascriptions.transition(Asc(d', t) |> rewrap)) {
-      | Some(d) =>
+    | Asc(d', t) =>
```

Comment:
This does ascription collapsing lazily. There's still an issue with substitution where if you substitute an indet ascription into another ascription they don't collapse.